staker/challenge-cache/cache.go

```diff
@@ -236,13 +230,6 @@ func readHashes(r io.Reader, numToRead uint64) ([]common.Hash, error) {
 		}
 		hashes = append(hashes, common.BytesToHash(buf))
 	}
```

Comment:
I think your observation is focused on just one of the possible problems with using a file-based cache. I don't *think* we'll actually end up in a situation where we need to deal with an empty file. But, I also think we should really be considering a more robust database-backed implementation in the future. I've opened NIT-2968 to track that possible future after further design consideration.

For now, I'm going to resolve this comment thread because I think we can launch with the system as it is.